content/blog/living-off-the-ipad-as-an-engineer/index.md

```diff
@@ -0,0 +1,110 @@
+
+A great way to make sure some coding work gets done when we get to travel again ✈️
+
+## Hardware Dev: Remote Server and Remote Desktop
+
+Here comes the harder part: hardware (*easy)*.
+
+To put it bluntly, there’s no SSHing into Computer Aided Design software. There’s no way of vim-editing a SolidWorks file. [OpenSCAD](https://www.openscad.org/) might be an exception to these statements, yet it’s most definitely niche.
+
+Two options I have explored:
+
+* use some kind of Remote Desktop software to access our closet computer/server and run the software remotely;
+
+* choose a web-based CAD software.
+
+The first option applies to a broad range of software beyond just CAD. The real bottleneck here is the quality of the iPad app. I’ve gone through many of the free options *à la* TeamViewer or Chrome Remote Desktop. Yet, none provided mandatory things for my use case like full mouse buttons support (including click-and-drag with the wheel, for instance). Jumping to the premium territory, Splashtop remains free for personal and same-network use, with a great iPad app, but has a monthly fee for real remote access. The one that ended up meeting all my needs was the $19.99 [Jump Destkop](https://jumpdesktop.com/). With its outdated app icon and steep price tag, this was clearly not my first choice, but their Fluid Remote Desktop protocol for Windows and macOS has just been the smoothest. It works wonders with external mice and Trackpads, plus has full external monitor support on the iPad with automatic resolution matching. On top of that, it supports VNC (even over SSH tunnels) to connect to Linux hosts such as a (local) Raspberry Pi or other instances I use for work daily.
+
+And it resisted the test of time: the whole design for my [GeeXY 3D printer project](https://www.notion.so/Geeetech-CoreXY-Conversion-GeeXY-b46d9f7b4b0643faa60bd2f20399c0b6) was created through Jump Desktop on the iPad. No regret so far.
+
+![My GeeXY printer model, fully created through Jump on the iPad](freecad.png)
+
+The second option lies in the cloud web-based CAD system: [Onshape](http://onshape.com). I’ve been bullish on their system through college since they’ve nailed quite a lot of CAD aspects that are out of this article’s scope. While I do use it from time to time on Safari on the iPad, I’ve been loyal to the great open-source [FreeCAD](https://freecadweb.org/) lately because I’m also using it at work (it even runs on Linux!).
+
+![Despite a few small bugs on Safari, Onshape is an excellent CAD system, fully usable on the iPad](onshape.png)
+
+A life of workarounds? Most definitely.
+
+Coming from macOS, the holy grail for many curious people who care about a furnished software library, first-class terminal experience and compatibility whatsoever, the fully mobile iPad path is definitely not the safest journey for our specific needs as developers or engineers.
```

Comment:
Coming from macOS, the holy grail for many curious people who care about a furnished software library, first-class terminal experience and compatibility, the fully mobile iPad path is definitely not the easiest path to meet our specific needs as developers or engineers.